Userland/Libraries/LibWeb/HTML/Navigable.cpp

```diff
@@ -493,7 +493,12 @@ static WebIDL::ExceptionOr<Optional<NavigationParams>> create_navigation_params_
             // FIXME: Let processEarlyHintsResponse be the following algorithm given a response earlyResponse:
 
             // Let processResponse be the following algorithm given a response fetchedResponse:
-            auto process_response = [&response](JS::NonnullGCPtr<Fetch::Infrastructure::Response> fetch_response) {
+            auto process_response = [&response, navigable, navigation_id](JS::NonnullGCPtr<Fetch::Infrastructure::Response> fetch_response) {
+                if (navigation_id.has_value() && (!navigable->ongoing_navigation().has<String>() || navigable->ongoing_navigation().get<String>() != *navigation_id)) {
+                    // NOTE: It's not specified, but we need to check if the navigation corresponding to the response has been canceled.
+                    //       If it has, we should return early to avoid modifying references pointing to a freed stack.
+                    return;
+                }
```

Comment:
This feels very sketchy. Is there a more fool-proof way we can prevent the `&response` capture from escaping this scope?